src/content/1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage.md

```diff
@@ -0,0 +1,227 @@
+  path: demo-cqrs-hook-usage/{customerId}/toggle-is-allowed-for-review
+  methods: [POST]
+  defaults:
+    _controller: 'DemoCQRSHooksUsage\Controller\Admin\CustomerReviewController::toggleIsAllowedForReviewAction'
+  requirements:
+    customerId: \d+
+```
+
+Route name `ps_democqrshooksusage_toggle_is_allowed_for_review` matches the one that was passed as mandatory option when creating the
+`ToggleColumn`. 
+
+#### Extending grid query builder
+
+By just extending grid definition we won't be able to display any data since we need to fetch it first. Luckily, we can extend grid query builder
+
+
+```php
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Search\Filters\CustomerFilters;
+
+/**
+     * Hook allows to modify Customers query builder and add custom sql statements.
+     *
+     * @param array $params
+     */
+    public function hookActionCustomerGridQueryBuilderModifier(array $params)
+    {
+        /** @var QueryBuilder $searchQueryBuilder */
+        $searchQueryBuilder = $params['search_query_builder'];
+
+        /** @var CustomerFilters $searchCriteria */
+        $searchCriteria = $params['search_criteria'];
+
+        $searchQueryBuilder->addSelect(
+            'IF(dcur.`is_allowed_for_review` IS NULL,0,dcur.`is_allowed_for_review`) AS `is_allowed_for_review`'
+        );
+
+        $searchQueryBuilder->leftJoin(
+            'c',
+            '`' . pSQL(_DB_PREFIX_) . 'democqrshooksusage_reviewer`',
+            'dcur',
+            'dcur.`id_customer` = c.`id_customer`'
+        );
+
+        if ('is_allowed_for_review' === $searchCriteria->getOrderBy()) {
+            $searchQueryBuilder->orderBy('dcur.`is_allowed_for_review`', $searchCriteria->getOrderWay());
+        }
+
+        foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
+            if ('is_allowed_for_review' === $filterName) {
+                $searchQueryBuilder->andWhere('dcur.`is_allowed_for_review` = :is_allowed_for_review');
+                $searchQueryBuilder->setParameter('is_allowed_for_review', $filterValue);
+
+                if (!$filterValue) {
+                    $searchQueryBuilder->orWhere('dcur.`is_allowed_for_review` IS NULL');
+                }
+            }
+        }
+    }
+
+```
+
+This sample demonstrates how to extend sql of the customers grid. From our custom database table `democqrshooksusage_reviewer` we fetch the result of field `is_allowed_for_review`. This name must match the id we added in the grid definition. In order for sorting to work we also add `orderBy` condition and finally, in order
+for filters to work `where` conditions are added if the filter exists in `$searchCriteria->getFilters()`.
+
+#### Result
+
+After completing the steps above by going to customers list you should see new column "allowed for review" added.
+
+{{< figure src="../img/extended_customers_grid.png" title="Allowed for review column added to customers list" >}}
+
```

Comment:
Please add a summary of the "operation":
- register new column in grid definition
- as this column allows to toggle a field using an ajax request, register a controller to handle this ajax request
- fetch the data required to populate the new column

Reply:
I don't think we need summary here due to:

1. Its not the end of whole topic - how to extend identifiable object form part is missing here.
2. This summary exactly repeats itself or repeats the topic headlines of which users will already be familiar then they come to this point - lets see if we need it after I finish whole PR

Reply:
Makes sense 👍